src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -277,7 +283,15 @@ ReactDOMComponent.Mixin = {
   mountComponent: function(rootID, transaction, context) {
     this._rootNodeID = rootID;
 
-    assertValidProps(this, this._currentElement.props);
+    var props = this._currentElement.props;
+
+    var wrapper = wrappers[this._tag];
+    if (wrapper) {
+      wrapper.mountWrapper.call(this, props, context);
```

Comment:
As cool as I think sweet.js is, it actually parses the code, which means it'll likely break on ES6/ES7 features it doesn't support, and on JSX. I think sweet.js and babel are mutually exclusive.